lib/streamlit/runtime/forward_msg_cache/memory_forward_msg_cache.py

```diff
@@ -279,6 +248,28 @@ def clear(self) -> None:
         """Remove all entries from the cache"""
         self._entries.clear()
 
+    def _create_reference_msg(self, msg: ForwardMsg) -> ForwardMsg:
+        if not msg.metadata.cacheable or not msg.hash:
+            raise RuntimeError("Cannot create reference for non-cacheable message!")
+        ref_msg = ForwardMsg()
+        ref_msg.forward_msg_ref.ref_url = f"{self.base_url}?hash={msg.hash}"
+        ref_msg.forward_msg_ref.ref_hash = msg.hash
+        ref_msg.forward_msg_ref.checksum_crc32 = crc32(msg.SerializeToString())
```

Comment:
nit: this might have some bigger compute overhead if the message is big. Probably something to that needs to be tested carefully. Maybe we can also only just reuse the (md5) hash to verify the message